Doppler.Sap/Services/BillingServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using Doppler.Sap.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Doppler.Sap.Services
+{
+    public static class ServicesCollectionExtensions
+    {
+        public static void AddServices(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.AddSingleton<DummyBillingService>();
+            services.AddTransient<BillingService>();
+            services.AddSingleton<DummyBusinessPartnerService>();
+            services.AddTransient<BusinessPartnerService>();
+
+            var section = configuration.GetSection("SapConfig");
+            var sapSettings = new SapConfig();
+            section.Bind(sapSettings);
+
+            services.AddSingleton<IBillingService>(provider =>
+            {
+                if (!section.Exists() || sapSettings.UseDummyData)
+                    return provider.GetRequiredService<DummyBillingService>();
+
+                return provider.GetRequiredService<BillingService>();
```

Comment:
If the configuration has been read before, we could move the condition out of this delegate, what is more, we could register only one service and all of this is not required.

What do you think about using the same pattern than in Billing API and postpone the reading of the configuration to when the service is resolved?

Reply:
ok, I am analyzing it and seeing how it was implemented in the billing api